Track unregistered stale connection count

SteveHNH opened the pull request:
RHCLOUD-9224

Signed-off-by: Stephen Adams <tsadams@gmail.com>

dehort commented on internal/controller/redis_connector.go:
```diff
@@ -70,6 +70,7 @@ func UnregisterWithRedis(client *redis.Client, account, nodeID, hostname string)
 	_, err := client.TxPipelined(func(pipe redis.Pipeliner) error {
 		client.Del(getConnectionKey(account, nodeID))
 		removeIndexes(client, account, nodeID, hostname)
+		metrics.unregisterStaleConnectionFromRedis.Inc()
```
This method is used in the case of normal un-registration so I don't think this is the right place for this counter.

dehort reviewed (changes requested):
For RHCLOUD-9224, I was thinking of modifying the connection cleaner util so that it recorded the stale connection cleanup metric.  I added a bit more details to RHCLOUD-9224.  Let me know if we need to discuss.

SteveHNH commented:
@dehort Give it a look now. I moved the metrics collection bit into connection_cleaner (like I was supposed to), and added prometheus push gateway config.

dehort commented on cmd/connection_cleaner/main.go:
```diff
@@ -33,18 +40,36 @@ func getRunningPods(dnsName string) (RunningPods, error) {
 	}
 }
 
+func pushValues(cfg *config.Config, connCount *Metrics) {
```
nit-picky:  What do you think about naming this method something a little bit more descriptive?  Something like pushMetricsToPrometheus()?

dehort commented on internal/config/config.go:
```diff
@@ -44,6 +44,7 @@ const (
 	GATEWAY_ACTIVE_CONNECTION_REGISTRAR_POLL_MAX_DELAY = "Gateway_Active_Connection_Registrar_Poll_Max_Delay"
 	GATEWAY_CLUSTER_SERVICE_NAME                       = "Gateway_Cluster_Service_Name"
 	NODE_ID                                            = "ReceptorControllerNodeId"
+	PROM_PUSH_GW                                       = "PromPushGW"
```
nit-picky:  Can you make this option name a bit more verbose?  Maybe "Prometheus_Push_Gateway"?

dehort commented on cmd/connection_cleaner/main.go:
```diff
@@ -33,18 +40,36 @@ func getRunningPods(dnsName string) (RunningPods, error) {
+
+	connCount := &Metrics{
+		unregisterStaleConnectionFromRedis: promauto.NewCounter(prometheus.CounterOpts{
+			Name: "receptor_controller_unregister_stale_connection_from_redis_count",
```
Can we change the name to "receptor_controller_connection_cleaner_unregister_stale_connection_from_redis_count"?  I think I'd like to track this count separately from the "usual" method of removing stale connections.

dehort reviewed (changes requested):
This looks good overall.  I left a couple of nit-picky comments.  Sorry, I favor being overly verbose in naming things.  I find it helpful when I come back to the code later down the road.  Let me know what you think!

dehort commented:
Does the push gateway require credentials?

SteveHNH commented:
> 
> 
> Does the push gateway require credentials?

As far as I know, it does not. We do need to update the network policy to allow our controller to hit it though.
